packages/middleware-sdk-transcribe-streaming/src/websocket-handler.spec.ts

```diff
@@ -6,35 +6,98 @@ import { PassThrough } from "stream";
 import { WebSocketHandler } from "./websocket-handler";
 
 describe("WebSocketHandler", () => {
+  const mockHostname = "localhost:6789";
+  const mockUrl = `ws://${mockHostname}/`;
+
   beforeEach(() => {
     (global as any).WebSocket = WebSocket;
   });
+
   afterEach(() => {
     WS.clean();
     jest.clearAllMocks();
   });
+
   it("should contain protocol metadata", () => {
     const handler = new WebSocketHandler();
     expect(handler.metadata.handlerProtocol).toEqual("websocket");
   });
 
+  it("populates socket in socket pool based on handle() requests", async () => {
+    const handler = new WebSocketHandler();
+    const server = new WS(mockUrl);
+
+    // @ts-expect-error Property 'sockets' is private and only accessible within class 'WebSocketHandler'.
+    expect(handler.sockets[mockUrl]).not.toBeDefined();
+
+    await handler.handle(
+      new HttpRequest({
+        body: new PassThrough(),
+        hostname: mockHostname,
+        protocol: "ws:",
+      })
+    );
+
+    // @ts-expect-error Property 'sockets' is private and only accessible within class 'WebSocketHandler'.
+    expect(handler.sockets[mockUrl]).toBeDefined();
+    // @ts-expect-error Property 'sockets' is private and only accessible within class 'WebSocketHandler'.
+    expect(handler.sockets[mockUrl].length).toBe(1);
+
+    await handler.handle(
+      new HttpRequest({
+        body: new PassThrough(),
+        hostname: mockHostname,
+        protocol: "ws:",
+      })
+    );
+
+    // @ts-expect-error Property 'sockets' is private and only accessible within class 'WebSocketHandler'.
+    expect(handler.sockets[mockUrl].length).toBe(2);
+  });
+
+  it("closes socket in socket pool on handler.destroy()", async () => {
+    const handler = new WebSocketHandler();
+    const server = new WS(mockUrl);
+
+    await handler.handle(
+      new HttpRequest({
+        body: new PassThrough(),
+        hostname: mockHostname,
+        protocol: "ws:",
+      })
+    );
+
+    // @ts-expect-error Property 'sockets' is private and only accessible within class 'WebSocketHandler'.
+    const socket = handler.sockets[mockUrl][0];
+
+    expect(socket.readyState).toBe(WebSocket.OPEN);
+    handler.destroy();
+
+    // Verify that socket.close() is called
+    expect(socket.readyState).toBe(WebSocket.CLOSING);
```

Comment:
on what timing does this transition to `WebSocket.CLOSED`?

Reply:
I tried timeouts of 1s and 5s, but the state was still closing.
That's why I updated source code to remove sockets with closing state, and verified state to be closing.